src/Service.Tests/Unittests/ConfigValidationUnitTests.cs

```diff
@@ -1747,5 +1755,189 @@ string[] linkingTargetFields
+        [DataRow(true, null, "Entity: EntityA has a null rest path. Accepted value types are: string, boolean.",
+            DisplayName = "NULL rest path configured for an entitiy fails config validation.")]
+        [DataRow(true, 1, $"Entity: EntityA has rest path specified with incorrect data type. Accepted data types are: string, boolean.",
+            DisplayName = "Rest path configured as integer for an entitiy fails config validation.")]
+        public void ValidateRestPathForEntityInConfig(
+            bool exceptionExpected,
+            object restPathForEntity,
+            string expectedExceptionMessage = "")
+        {
+            Dictionary<string, Entity> entityCollection = new();
+
+            // Create first entity with REST settings.
+            Entity entity = SchemaConverterTests.GenerateEmptyEntity();
+            entity.Rest = new RestEntitySettings(Path: restPathForEntity);
+            entityCollection.Add("EntityA", entity);
+
+            RuntimeConfig runtimeConfig = new(
+                Schema: "UnitTestSchema",
+                DataSource: new DataSource(DatabaseType: DatabaseType.mssql),
+                RuntimeSettings: AuthorizationHelpers.CreateRuntimeSettings(),
+                Entities: entityCollection
+                );
+            runtimeConfig.DetermineGlobalSettings();
+            if (exceptionExpected)
+            {
+                DataApiBuilderException dabException =
+                    Assert.ThrowsException<DataApiBuilderException>(() => RuntimeConfigValidator.ValidateEntityConfiguration(runtimeConfig));
+                Assert.AreEqual(expectedExceptionMessage, dabException.Message);
+                Assert.AreEqual(expected: HttpStatusCode.ServiceUnavailable, actual: dabException.StatusCode);
+                Assert.AreEqual(expected: DataApiBuilderException.SubStatusCodes.ConfigValidationError, actual: dabException.SubStatusCode);
+            }
+            else
+            {
+                RuntimeConfigValidator.ValidateEntityConfiguration(runtimeConfig);
+            }
+        }
+
+        /// <summary>
+        /// Test to validate that when multiple entities have the same custom rest path configured, we throw an exception.
+        /// </summary>
+        /// <param name="exceptionExpected">Whether an exception is expected as a result of test run.</param>
+        /// <param name="restPathForFirstEntity">Custom rest path to be configured for the first entity.</param>
+        /// <param name="restPathForSecondEntity">Custom rest path to be configured for the second entity.</param>
+        /// <param name="expectedExceptionMessage">The expected exception message.</param>
+        [DataTestMethod]
+        [DataRow(false, "restPathA", "restPathB", DisplayName = "Unique rest paths configured for entities pass config validation.")]
+        [DataRow(false, true, false, DisplayName = "Rest path configured as boolean values for an entities pass config validation.")]
```

Comment:
Can you please double check our official configuration file docs/json schema if we allow boolean for path. I thought it was only allowed for `rest`

Reply:
You're correct. Will do the required change.